src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -346,12 +346,15 @@ string TruncateIfNecessary(string text)
 
     static bool TryGetDictionary(object value, Type valueType, [NotNullWhen(true)] out IDictionary? dictionary)
     {
-        if (valueType.IsConstructedGenericType &&
-            valueType.GetGenericTypeDefinition() == typeof(Dictionary<,>) &&
-            IsValidDictionaryKeyType(valueType.GenericTypeArguments[0]))
+        if (valueType.IsConstructedGenericType)
         {
-            dictionary = (IDictionary)value;
-            return true;
+            var definition = valueType.GetGenericTypeDefinition();
```

Comment:
would it be better to do an `is IDictionary` first, and then (if not) avoid the GetGenericTypeDefinition and GenericTypeArguments  ?

Reply:
done